Website/htdocs/mpmanager/app/Http/Controllers/Api/PaymentHistoryController.php

```diff
@@ -56,6 +56,38 @@ public function show(int $payerId, string $period, $limit = null, $order = 'ASC'
 
         $payments = (new PaymentHistory)->getFlow('person', $payerId, $period, $limit, $order);
 
+
+        $flowList = [];
+
+        foreach ($payments as $payment) {
+            $flowList[$payment['aperiod']][$payment['payment_type']] = $payment['amount'];
+        }
+
+        return $flowList;
+    }
+    public function showForAgentCustomers( string $period, $limit = null, $order = 'ASC')
```

Comment:
`showForAgentCustomers` is basically the copy of the `show` method. Please define a separate method for the commonly used code